crates/paralegal-policy/src/context.rs

```diff
@@ -40,10 +40,118 @@ pub enum EdgeType {
     DataAndControl,
 }
 
+/// Data type representing nodes in the SPDG for a particular controller.
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Node<'a> {
+    /// ControllerId of the node.
+    pub ctrl_id: ControllerId,
+    /// The data of the node.
+    pub node: NodeType<'a>,
```

Comment:
For naming consistency I'd call this `typ` or `type_` now.

Reply:
done